### PR TITLE
Calc balance & money pile total on save.

### PR DESCRIPTION
**Pull request by cghislai**
Also set the time.

### CODE REVIEW

File: comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountService.java [hunk 1]
```diff
@@ -235,4 +264,26 @@ public void cancelOpenBalance(Balance balance) {
         entityManager.remove(managedBalance);
     }
 
+    private Balance calcBalance(Balance balance) {
+        List<MoneyPile> moneyPileList = findMoneyPileListForBalance(balance);
+
+        double totalBalance = moneyPileList.stream()
+                .map((moneyPile) -> moneyPile.getTotal())
+                .mapToDouble((moneyPileTotalBigDecimal) -> moneyPileTotalBigDecimal.doubleValue())
```
Comment (ymajoros) [inline comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountService.java]: No double please, we use fixed point arithmetic everywhere (BigDecimal).

File: comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountService.java [hunk 2]
```diff
@@ -235,4 +264,26 @@ public void cancelOpenBalance(Balance balance) {
...
+    }
+
+    private MoneyPile calcMoneyPile(MoneyPile moneyPile) {
+        BigDecimal count = moneyPile.getCount();
+        BigDecimal unitAmount = moneyPile.getUnitAmount();
+        BigDecimal totalValue = count.multiply(unitAmount);
+        moneyPile.setTotal(totalValue);
+        MoneyPile managedMoneyPile = entityManager.merge(moneyPile);
+
+        calcBalance(managedMoneyPile.getBalance());
```
Comment (ymajoros) [inline comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountService.java]: introduce var

Comment (ymajoros) [inline comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountService.java]: Please do something with result or don't return anything.

**Comment by ymajoros** [comment]
almost :-1:

**Comment by cghislai** [comment]
Updated.
Sorry about that.

File: comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountingUtils.java [hunk 3]
```diff
@@ -90,4 +92,21 @@ private static SalePrice combineSalePrices(SalePrice salePrice1, SalePrice saleP
...
+        BigDecimal count = moneyPile.getCount();
+        BigDecimal unitAmount = moneyPile.getUnitAmount();
+        BigDecimal totalValue = count.multiply(unitAmount);
+        moneyPile.setTotal(totalValue);
+        return moneyPile;
+    }
+    
+    public static Balance calcBalance(Balance balance, List<MoneyPile> moneyPiles) {
+        BigDecimal balanceValue = moneyPiles.stream()
+                .map((moneyPile) -> moneyPile.getTotal())
```
Comment (ymajoros) [inline comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountingUtils.java]: .map(moneyPile::getTotal)

**Comment by ymajoros** [comment]
Don't be sorry for my maniacal preferences ;-)

Merged with minor comment.

**Pull request merged**